js/src/Map.js

```diff
@@ -244,6 +277,14 @@ export class LeafletMapView extends utils.LeafletDOMWidgetView {
       this.create_panes();
       this.layer_views.update(this.model.get('layers'));
       this.control_views.update(this.model.get('controls'));
+
+      var layer_list = this.model.get('layers');
+      for (let i = 1; i < layer_list.length; i++) { // starting index is 1, since we are not taking into account the TileLayer at index 0
+        var subitem_list = layer_list[i].attributes.subitems
+        this.subitem_views.update(subitem_list);
```

Comment:
It looks like this doesn't work if multiple layers have subitems? Only the latest added layer will add its subitems?

You probably need to call `this.subitem_views.update` with the concatenation of all subitems of all layers.

Reply:
We tried to implement what is proposed in this comments. When adding 2 layers having their own subitems, they all displayed on the same map. Nevertheless there remains an issue with the method remove_view. `Map.js:221 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'includes')
    at LeafletMapView.remove_subitem_view (Map.js:221:1) `